src/core/security.tf

```diff
@@ -231,3 +231,76 @@ resource "azurerm_key_vault_secret" "jwt_public_key" {
 
   key_vault_id = module.key_vault.id
 }
+
+resource "azurerm_key_vault_certificate" "jwt_certificate" {
+  name         = "jwt-crt"
+  key_vault_id = module.key_vault.id
+
+  certificate {
+    contents = pkcs12_from_pem.jwt_pkcs12.result
+    password = ""
+  }
+
+  # to be provided even if not required due to issue https://github.com/hashicorp/terraform-provider-azurerm/pull/14225
+  certificate_policy {
+    issuer_parameters {
+      name = "Self"
+    }
+
+    key_properties {
+      exportable = true
+      key_size   = 2048
+      key_type   = "RSA"
+      reuse_key  = true
+    }
+
+    secret_properties {
+      content_type = "application/x-pkcs12"
+    }
+  }
+
+  tags = var.tags
+}
+
+resource "null_resource" "upload_jwks" {
+  triggers = {
+    "changes-in-jwt" : azurerm_key_vault_certificate.jwt_certificate.thumbprint
+  }
+  provisioner "local-exec" {
+    command = <<EOT
+              mkdir -p "${path.module}/.terraform/tmp"
+              pip install authlib
+              jwk=$(python "${path.module}/utils/py/jwkFromPem.py" "${format("%s\n%s\n%s", "-----BEGIN CERTIFICATE-----", replace(azurerm_key_vault_certificate.jwt_certificate.certificate_data_base64, "/(.{64})/", "$1\n"), "-----END CERTIFICATE-----")}")
```

Comment:
is possibile to use openssl commands?

Reply:
it's difficutl build the required infos (modulus and exponent of the rsa algorithm encoded as asn1) using openssl. this python script is the simplest method found

Reply:
This approach doesn't take into consideration a potential rotation and coexistence of more keys for a time window.
During a rotation the JWKS will be a set of JWK (new and old key), the code will use the first key to sign new tokens, while clients still holding older tokens will be able to talk to the server (old key still there).

Reply:
in the new logic, the jwks will take into account the previous file